v0/destinations/klaviyo/util.js

```
@@ -0,0 +1,175 @@
+    targetUrl = `${targetUrl}/subscribe`;
+    profile.sms_consent = get(traitsInfo.properties, "smsConsent")
+      ? get(traitsInfo.properties, "smsConsent")
+      : destination.Config.smsConsent;
+    profile.$consent = get(traitsInfo.properties, "consent")
+      ? get(traitsInfo.properties, "consent")
+      : destination.Config.consent;
+  }
+  profile = removeUndefinedValues(profile);
+  const payload = {
+    profiles: [profile]
+  };
+  const response = defaultRequestConfig();
+  response.endpoint = targetUrl;
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.params = { api_key: destination.Config.privateApiKey };
+  response.headers = {
+    "Content-Type": "application/json"
+  };
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+
+  return response;
+};
+
+/**
+ * This function is used to check if the user needs to be added to list or needs to be subscribed or not.
+ * Building and returning response array for both the members(for adding to the list) and subscribe
+ * endpoints (for subscribing)
+ * @param {*} message 
+ * @param {*} traitsInfo 
+ * @param {*} destination 
+ * @returns 
+ */
+const checkForMembersAndSubscribe = (message, traitsInfo, destination) => {
+  const responseArray = [];
+  if (
+    (!!destination.Config.listId || !!get(traitsInfo.properties, "listId")) &&
+    destination.Config.privateApiKey
+  ) {
+    const membersResponse = addUserToList(
+      message,
+      traitsInfo,
+      LIST_CONF.MEMBERSHIP,
+      destination
+    );
+    responseArray.push(membersResponse);
+    if (get(traitsInfo.properties, "subscribe") === true) {
+      const subscribeResponse = addUserToList(
+        message,
+        traitsInfo,
+        LIST_CONF.SUBSCRIBE,
+        destination
+      );
+      responseArray.push(subscribeResponse);
+    }
+  } else {
+    logger.info(
+      `Cannot process list operation as listId is not available, either in message or config, or private key not present`
+    );
```

Lets throw error here (other wise we will fail silently)